samcli/lib/utils/definition_validator.py

```diff
@@ -48,13 +48,27 @@ def validate(self) -> bool:
             If detect_change is set, False will also be returned if there is
             no change compared to the previous validation.
         """
+        old_data = self._data
+
+        if not self.raw_validate():
+            return False
+        return old_data != self._data if self._detect_change else True
+
+    def raw_validate(self) -> bool:
```

Comment:
I feel like this method should be named as `validate` and the actual `validate` method should be something like `check_change`, what do you think?

Reply:
Yes makes sense. `validate_template` and `validate_change` would make sense to me.